app/controllers/api/v2/moves_actions.rb

```diff
@@ -92,6 +103,15 @@ def common_move_attributes
       end
     end
 
+    def validate_move_status
+      status = move_params.fetch(:attributes, {})[:status]
+      if status.present?
+        validator = Moves::StatusValidator.new(status: status, cancellation_reason: move_params.fetch(:attributes, {})[:cancellation_reason], rejection_reason: move_params.fetch(:attributes, {})[:rejection_reason])
+        raise ActiveModel::ValidationError, validator unless validator.valid?
+      end
+      status
+    end
+
```

Comment:
This logic looks like it's repeated in two other places with the difference being which object we take status from. Is there a case for abstracting it to keep it consistent? Or is it the same now, but that's by chance and we'd expect them to diverge in the future?

Reply:
Actually, ignore me! I can see the validation logic itself is abstracted!

Reply:
😅